pkg/test/kube/kubectl.go

```diff
@@ -81,17 +81,32 @@ func (c *kubectl) apply(namespace string, filename string) error {
 
 func (c *kubectl) applyInternal(namespace string, files []string) error {
 	for _, f := range files {
-		command := fmt.Sprintf("kubectl apply %s %s -f %s", c.configArg(), namespaceArg(namespace), f)
-		scopes.CI.Infof("Applying YAML: %s", command)
-		s, err := shell.Execute(true, command)
-		if err != nil {
-			scopes.CI.Infof("(FAILED) Executing kubectl: %s (err: %v): %s", command, err, s)
-			return fmt.Errorf("%v: %s", err, s)
+		if !isEmpty(f) {
+			command := fmt.Sprintf("kubectl apply %s %s -f %s", c.configArg(), namespaceArg(namespace), f)
+			scopes.CI.Infof("Applying YAML: %s", command)
+			s, err := shell.Execute(true, command)
+			if err != nil {
+				scopes.CI.Infof("(FAILED) Executing kubectl: %s (err: %v): %s", command, err, s)
+				return fmt.Errorf("%v: %s", err, s)
+			}
 		}
 	}
 	return nil
 }
 
+func isEmpty(f string) bool {
+	fileInfo, err := os.Stat(f)
+	if err != nil {
+		scopes.CI.Warnf("Error stating file %s prior to applying: %v", f, err)
+		return true
+	}
+	if fileInfo.Size() == 0 {
+		scopes.CI.Warnf("Unable to apply empty YAML file: %s", f)
```

Comment:
if this is a utility function, then I dont think you need these printfs.. esp since the caller is going to print out the errors. It just makes the logs a bit more chatty dont you think?

Reply:
The warning doesn’t have to be there but it makes it easier to debug if you create an empty yaml by accident.